src/ratingservice/main.py

```diff
@@ -77,6 +78,43 @@ def makeResult(data):
 #
 
 
+@app.route('/ratings', methods=['GET'])
+def getRatings():
+    '''Gets a list of all ratings.
+
+    Returns:
+        HTTP status 200 and Json payload { ratings: [{'id': (string), 'rating': (number)}] }
+        HTTP status 500 when there is an error querying DB or no data
+    '''
+
+    conn = getConnection()
+    if conn == None:
+        return makeError(500, 'failed to connect to DB')
+    try:
+        with conn.cursor() as cursor:
+            cursor.execute("SELECT eid, ROUND(rating,4) FROM ratings")
+            result = cursor.fetchall()
+        conn.commit()
+        print("1")
+        if result != None:
+            print("1")
+            ratings = []
+            print("1")
+            for row in result:
+                # cast to float because flas.jsonify doesn't work with decimal
+                ratings.append({"id": row[0].strip(), "rating": float(row[1])})
+            print("1")
+            return makeResult({
+                'ratings': ratings,
+            })
+        else:
+            return makeError(500, 'No available ratings')
+    except:
```

Comment:
are there specific exceptions you can catch here?

Reply:
it is generally catches database errors starting with connection error and ending with sql and casting. i do not know if more detailed handling is required. prefer to leave it generic at the moment.

Reply:
generic catches are a bad practice, you should really add some of the exceptions you expect to catch here. Right now your block will catch exceptions like `KeyboardInterrupt` and `SystemExit`, which could cause some strange behaviours

Reply:
i agree. with that it is not a real application and there is no practical reason to threat each exception separately.

Reply:
I'd rather invest the time now in code review, rather than more time later to hunt down bugs.

This is the [exception inheretance tree for psycopg2](https://www.psycopg.org/docs/module.html):
```
Exception
|__ Warning
|__ Error
    |__ InterfaceError
    |__ DatabaseError
        |__ DataError
        |__ OperationalError
        |__ IntegrityError
        |__ InternalError
        |__ ProgrammingError
        |__ NotSupportedError
```

The documentation says it's sufficient to just catch [`psycopg2.Error`](https://www.psycopg.org/docs/module.html#psycopg2.Error) to handle any psycopg2 errors. If you really want to be safe, you could even just catch the generic Python `Exception`.  That's also a bad practice, but it would at least make sure the program won't swallow exit signals

Reply:
Thanx. I will use DatabaseError then.